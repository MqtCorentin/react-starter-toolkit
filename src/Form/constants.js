import { ROUTE_URL_FIELDS_CLASSICS } from 'Form/FieldsClassics/constants';

export const TITLE_BAR = 'Cr√©ation de formulaire';
export const TITLE = 'Liste des champs';
export const ROUTE_URL_FORM = '/form';

export const FIELDS = [
  {
    id: ROUTE_URL_FIELDS_CLASSICS,
    title: 'Champ de saisie',
    storybook: {
      text: 'Storybook : Text Input',
      path: 'form-input-text--text',
    },
    designSystem: {
      text: 'Guidelines : Text Input',
      path: '/molecules/form-text/',
    },
    picture: {
      name: 'champ-texte.jpg',
      alt: 'champ texte',
    },
  },
];
