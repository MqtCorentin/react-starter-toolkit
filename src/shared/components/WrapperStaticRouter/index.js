import { StaticRouter } from 'react-router-dom/server';

export default ({ context = {}, children }) => <StaticRouter context={context}>{children}</StaticRouter>;
