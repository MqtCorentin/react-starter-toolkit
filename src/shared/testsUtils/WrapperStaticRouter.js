import React from 'react';
import { StaticRouter } from 'react-router-dom/server';

export default ({ context = {}, children }) => <StaticRouter context={context}>{children}</StaticRouter>;
