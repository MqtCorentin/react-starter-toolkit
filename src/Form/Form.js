import React from 'react';
import PropTypes from 'prop-types';
import Panel, { panelPropTypes } from 'shared/components/Panel';
import { FIELDS, TITLE_BAR, TITLE } from './constants';

export const Panels = ({ fields }) =>
  fields.map(({ id, ...rest }) => (
    <div key={id} className="col col-sm-12 col-md-12 col-lg-4 col-xl-4">
      <Panel {...rest} />
    </div>
  ));

const FormPage = ({ header, footer, title, menu, fields }) => (
  <>
    {header()}
    {menu()}
    {title({ title: TITLE_BAR })}
    <div className="af-main container">
      <h2 className="af-title--content">{TITLE}</h2>
      <Panels fields={fields} />
    </div>
    {footer()}
  </>
);

export const formPagePropTypes = {
  fields: PropTypes.arrayOf(
    PropTypes.shape({
      ...panelPropTypes,
    }),
  ),
};

export const formPageDefaultProps = {
  fields: FIELDS,
};

FormPage.propTypes = formPagePropTypes;
FormPage.defaultProps = formPageDefaultProps;

FormPage.propTypes = formPagePropTypes;

export default FormPage;
