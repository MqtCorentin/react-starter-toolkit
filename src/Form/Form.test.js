import React from 'react';
import WrapperStaticRouter from 'shared/components/WrapperStaticRouter';
import { render, act } from '@testing-library/react';
import Form from './Form';

const defaultProps = {
  header: () => {},
  title: () => {},
  footer: () => {},
  menu: () => {},
};
describe('<Form page/>', () => {
  it('Renders Form page component without crashing', async () => {
    await act(async () => {
      const { asFragment } = render(<Form {...defaultProps} />, { wrapper: WrapperStaticRouter });
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
