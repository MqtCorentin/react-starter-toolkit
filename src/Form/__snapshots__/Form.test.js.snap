// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form page/> Renders Form page component without crashing 1`] = `
<DocumentFragment>
  <div
    class="af-main container"
  >
    <h2
      class="af-title--content"
    >
      Liste des champs
    </h2>
    <div
      class="col col-sm-12 col-md-12 col-lg-4 col-xl-4"
    >
      <article
        class="af-panel"
      >
        <img
          alt="loading..."
          src="puff.svg"
          style="position: absolute; transform: translate(-50%,-50%); left: 50%; top: 50%;"
        />
        <header
          class="af-panel__header"
        >
          <h3
            class="af-panel__title"
          >
            Champ de saisie
          </h3>
        </header>
        <section
          class="af-panel__content"
          style="min-height: 200px; justify-content: center; display: flex; align-items: center;"
        >
          <a
            class="btn af-btn--circle"
            href="https://axaguildev.github.io/react-toolkit/latest/design//molecules/form-text/"
            rel="noreferrer"
            target="_blank"
            title="Storybook : Text Input"
          >
            <i
              class="glyphicon glyphicon-link"
            />
          </a>
          <a
            class="btn af-btn--circle"
            href="https://axaguildev.github.io/react-toolkit/latest/storybook/?path=/story/form-input-text--text"
            rel="noreferrer"
            target="_blank"
            title="Storybook : Text Input"
          >
            <i
              class="glyphicon glyphicon-link"
            />
          </a>
        </section>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
